cfy_manager/components/validations.py

```diff
@@ -296,29 +296,38 @@ def _check_ssl_file(input_name, kind='Key', password=None):
         password_err = ''
         if password:
             password_err = '(or the provided password is incorrect)'
-        raise ValidationError('{0} file {1} (input name: {2}) is invalid {3}'
-                              .format(kind, filename, input_name,
-                                      password_err))
+        raise ValidationError('{0} file {1} password is invalid {2}'
+                              .format(kind, filename, password_err))
 
 
-def _check_key_and_cert(prefix):
-    """Check that the provided cert and key actally match"""
-    cert_input = '{0}_cert_path'.format(prefix)
-    key_input = '{0}_key_path'.format(prefix)
-    cert_filename = config[SSL_INPUTS][cert_input]
-    key_filename = config[SSL_INPUTS][key_input]
-    password_input = '{0}_key_password'.format(prefix)
-    password = config[SSL_INPUTS].get(password_input)
+def check_certificates(component,
+                       cert_path='cert_path', key_path='key_path',
+                       ca_path='ca_path', key_password='key_password'):
+    """Check that the provided cert, key, and CA actally match"""
+    cert_filename = config[component].get(cert_path)
+    key_filename = config[component].get(key_path)
+
+    ca_filename = config[component].get(ca_path)
```

Comment:
ca_filename should fallback to `ssl_inputs: ca_cert_path` otherwise the `use_supplied_certificates` function will fail if `ca_path` is not provided by `cert_path` does in `rabbitmq` or any future components using this function

Reply:
In the case that a self-signed certificate is supplied, that behaviour would cause it to break.
I think this way of doing it is reasonable- if you're using the same CA cert for <component> then you supply the path to it to that component?

Otherwise we'd need some sort of conditional fall-through, which seems like it'd be more likely to add bugs than avoid problems?

Reply:
But then it means you need to place the path to the CA cert in every component even if it's the same across all of them.
Maybe the fallback should behave a bit differently, but I'm not really comfortable with "forcing" the user to place a `ca_path` on every component even thought it's the same across all of them.